hyper/cli.py

```diff
@@ -160,13 +161,16 @@ def set_request_data(args):
         elif i.sep == SEP_QUERY:
             params[i.key] = i.value
         elif i.sep == SEP_DATA:
-            body[i.key] = i.value
+            value = i.value
+            if is_py2:  # pragma: no cover
+                value = value.decode(PREFERRED_ENCODING)
+            body[i.key] = value
 
     if params:
         args.url.path += '?' + urlencode(params)
 
     if body:
-        content_type = 'application/json; charset=%s' % FILESYSTEM_ENCODING
+        content_type = 'application/json; charset=utf-8'
```

Comment:
For reference: https://tools.ietf.org/html/rfc7159#section-8.1 and https://tools.ietf.org/html/rfc7159#section-11. You'll note in the latter that there are no optional parameters (which is what `charset` is).